sys/arduino/include/spiport.hpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_arduino_api
+ * @{
+ *
+ * @file
+ * @brief       Definition of the Arduino 'SPI' interface
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#ifndef SPIPORT_H
+#define SPIPORT_H
+
+extern "C" {
+#include "arduino_board.h"
+#include "byteorder.h"
+#include "periph/spi.h"
+}
+
+/**
+ * @name    Arduino compatible SPI modes
+ * @{
+ */
+#define SPI_MODE0       (0)     /**< CPOL=0, CPHA=0 */
+#define SPI_MODE1       (1)     /**< CPOL=0, CPHA=1 */
+#define SPI_MODE2       (2)     /**< CPOL=1, CPHA=0 */
+#define SPI_MODE3       (3)     /**< CPOL=1, CPHA=1 */
+/** @} */
+
+/**
+ * @name    Arduino compatible bit order values for SPI
+ * @{
+ */
+#define MSBFIRST        (0)     /**< most significat bit first */
+/** @} */
+
+/**
+ * @brief   Arduino SPI configuration interface
+ */
+class SPISettings {
+    public:
+    /**
+     * @brief   Create a new SPI settings instance
+     *
+     * @param   clock       SPI clock in Hz to use
+     * @param   bitOrder    Has to be `MSBFIST`, for compatibility only
+     * @param   dataMode    SPI mode to use
+     */
+    SPISettings(uint32_t clock, uint8_t bitOrder, uint8_t dataMode);
+
+    /**
+     * @brief   Create a new SPI settings instance with default settings
+     */
+    SPISettings() : SPISettings(4000000, MSBFIRST, SPI_MODE0) { }
```

Comment:
Delegating constructors is only available with C++11. Either we add
```
CXXEXFLAGS += -std=c++11
```
to `sys/arduino/Makefile.include` or it has to be changed. Just try:
```
USEMODULE='periph_spi' make BOARD=arduino-mega2560 -C examples/arduino_hello-world
```

Reply:
My student also had the issue with missing C++11 support. This seems to only happen on AVR with the Ubuntu packages, which contains an horrible out of date version of avr-gcc. Since some years C++11 support is now enabled by default...

But I guess most users of RIOT are using Ubuntu (or at least enough to not simple ignore them). I think that Arduino is providing C++11 in any case, so Arudino libraries will also depend on that. I guess I'll add -std=c++11 simply when the `arduino` module is used.